libcasr/src/rust.rs

```diff
@@ -26,3 +29,300 @@ impl Exception for RustPanic {
         )))
     }
 }
+
+/// Structure provides an interface for processing stacktraces from RUST_BACKTRACE.
+pub struct RustStacktrace;
+
+impl ParseStacktrace for RustStacktrace {
+    fn extract_stacktrace(stream: &str) -> Result<Vec<String>> {
+        let lines = stream
+            .split('\n')
+            .map(|l| l.to_string())
+            .collect::<Vec<String>>();
+
+        let Some(backtrace_idx) = lines
+            .iter()
+            .enumerate()
+            .position(|(i, line)| line.starts_with("stack backtrace:") && i < lines.len() - 1)
+        else {
+            return Err(Error::Casr(
+                "Couldn't find start of stacktrace in Rust panic output".to_string(),
+            ));
+        };
+
+        let lines = &lines[backtrace_idx + 1..];
+        let re_func = Regex::new(r" *\d:.+0x[0-9a-f]+ - .*").unwrap();
+        let re_src = Regex::new(r" *at .+").unwrap();
+        let mut stacktrace = Vec::new();
+        let mut st_line = String::new();
+        for line in lines {
```

Comment:
Maybe we can use only one regex to extract entire stacktrace in one step `"( *\d+: *?0x[0-9a-f]+ - .*\n(?: *at .*\n)?)+"`